control-plane/plugin/src/resources/snapshot.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{
+    operations::GetSnapshots,
+    resources::{utils, SnapshotId, VolumeId},
+    rest_wrapper::RestClient,
+};
+use async_trait::async_trait;
+
+use crate::resources::utils::{CreateRow, GetHeaderRow};
+use prettytable::Row;
+
+/// A collection of VolumeSnapshot resource for all volumes.
+#[derive(clap::Args, Debug)]
+pub struct VolumesSnapshots {}
+
+/// A collection of VolumeSnapshot resource for a given volume.
+#[derive(clap::Args, Debug)]
+pub struct VolumeSnapshots {}
+
+#[derive(Debug, Clone, clap::Args)]
+/// BlockDevice args
+pub struct VolumeSnapshotArgs {
+    #[clap(long)]
+    /// Uuid of the volume (Optional).
+    volume: Option<VolumeId>,
+    #[clap(long)]
+    /// Uuid of the snapshot (Optional).
+    snapshot: Option<SnapshotId>,
+}
+
+impl VolumeSnapshotArgs {
+    /// get the volume id
+    pub fn volume(&self) -> Option<VolumeId> {
```

Comment:
Do they need to be pub? If so then, should we follow the doc comment convention. Start with Capital end with `.`

Reply:
Same for below?

Reply:
modified doc comments

Reply:
Why are these pub?

Reply:
Youn can, with pub(crate) but actually just remembered they need to be pub because the kubectl-mayastor plugin is external, so has to be pub.